### DISCUSSION TITLE
Add a specific library to the target libraries within a container

### DISCUSSION THREAD

**Discussion by myz-dev** [cross-rs/cross#706] [Q&A] [answered]
Hello everyone,

I am new to docker and everything related. I have tried to follow all instructions provided on the `README.md` in the main repository but my compile attempts still fail.
What I want to do: Cross compile my application for the `i686-unknown-linux-gnu` target.
When I run ` cross build -r --target i686-unknown-linux-gnu` I get the error, that a specific c library could not be found (`-lsqlite3` in my case). So everything up to the point where the container tries to link to -lsqlite3 seems to be working.
Now I want to add that library to the container.
To do that I created my own container based on the  `i686-unknown-linux-gnu`  parent. It looks like this:
```docker
FROM ghcr.io/cross-rs/i686-unknown-linux-gnu:edge

RUN  apt-get update && \
    apt-get install --assume-yes sqlite3 && \
    apt-get install --assume-yes libsqlite3-dev:i386\
```
The container runs the last command, finds the package and complains that its architecture is different from `i386`.
I think I get whats happening: The container itself is running a 64bit architecture (amd64) and rejects the i386 packages. 
So where in the container are the target's libraries installed? And how do I get a specific package installed in this directory?

Thank you very much.

**Comment by Emilgardis** [comment, marked as answer]
try running `dpkg --add-architecture i386` before install

**Reply by myz-dev** [reply]
Thank you so much, that did the trick! Thank you for this great project that makes cross compiling incredibly easy!
